ann_benchmarks/runner.py

```diff
@@ -108,42 +110,73 @@ def run(definition, dataset, count, run_count, batch):
     X_test = numpy.array(D['test'])
     distance = D.attrs['distance']
     print('got a train set of size (%d * %d)' % (X_train.shape[0], dimension))
-    print('got %d queries' % len(X_test))
 
     X_train, X_test = dataset_transform(D)
+    # X_train = X_train[:5000]
 
     try:
         prepared_queries = False
         if hasattr(algo, "supports_prepared_queries"):
             prepared_queries = algo.supports_prepared_queries()
 
-        t0 = time.time()
-        memory_usage_before = algo.get_memory_usage()
-        algo.fit(X_train)
-        build_time = time.time() - t0
-        index_size = algo.get_memory_usage() - memory_usage_before
-        print('Built index in', build_time)
-        print('Index size: ', index_size)
+        if not test_only:
+            per_client = len(X_train) // num_clients
+            offset = per_client * (id - 1)
+            if num_clients == id:
+                fit_args = [X_train[offset:]]
+            else:
+                fit_args = [X_train[offset:(offset + per_client)]]
+            print('inserting %d out of %d vectors' % (len(fit_args[0]), len(X_train)))
+            if "offset" and "cap" in inspect.getfullargspec(algo.fit)[0]:
+                fit_args.append(offset)
+                fit_args.append(len(X_train))
+            
+            t0 = time.time()
+            memory_usage_before = algo.get_memory_usage()
+            algo.fit(*fit_args)
+            build_time = time.time() - t0
+            index_size = algo.get_memory_usage() - memory_usage_before
+            print('Built index in', build_time)
+            print('Index size: ', index_size)
+        #     if build_only:
+        #         fn = get_result_filename(dataset, count)
+        #         fn = os.path.join(fn, definition.algorithm)
+        #         if not os.path.isdir(fn):
+        #             os.makedirs(fn)
+        #         fn = os.path.join(fn, f'build_stats_{id}.hdf5')
+        #         f = h5py.File(fn, 'w')
+        #         f.attrs["build_time"] = build_time
+        #         f.attrs["index_size"] = index_size
+        #         f.close()
+        else:
+            fn = get_result_filename(dataset, count)
+            fn = os.path.join(fn, 'build_stats.hdf5')
+            f = h5py.File(fn, 'r')
+            build_time = f.attrs["build_time"]
+            index_size = f.attrs["index_size"]
+            f.close()
```

Comment:
why do you read the file here?

Reply:
in the complete (build and test) run, these values are written to the results file. I'm not sure why we need them, but this way I pass them if we run in test only mode. these values are problematic if we don't run in split mode, or with more than one client, and should be calculated "outside" the clients scope anyway.
Maybe I can leave the build stats in separate file only and not have them in the results files